fedimintd/src/lib.rs

```diff
@@ -11,6 +11,9 @@ use crate::encrypt::{encrypted_read, encrypted_write};
 pub mod encrypt;
 pub mod ui;
 
+/// Version of the server code (should be the same among peers)
+pub const CODE_VERSION: &str = env!("GIT_HASH");
```

Comment:
Should this be `CARGO_PKG_VERSION_MAJOR` if we want to denote minor upgrades as non-consensus breaking?

Reply:
Yes, as soon we do releases that would be good. I wonder if we can modify the nix build code such that we can optionally set a release flag and it includes the version number instead of the hash? @dpc

Reply:
I don't think it has to go through nix. env!(`CARGO_PKG_VERSION_MAJOR`) is set by cargo/rustc at compile time, and it doesn't change too often. Hash is more difficult because it changes on every build, so we needed to move it to post-processing phase.

Reply:
I was thinking of having a flag that tells our nix build that it's a release and to include the SemVer version instead of a hash. I think that needs to be set via the environment?

We can have certain compatibility guarantees for releases (that we don't give PR to PR). So if there is a SemVer in that field we can determine compatibility more intelligently than when we put a hash in there.

Reply:
> I was thinking of having a flag that tells our nix build that it's a release and to include the SemVer version instead of a hash. I think that needs to be set via the environment?

What I've seen project do and tried myself, is to use `cargo-release` and keep the `version = ...` in the Cargo.toml`s include `-pre` suffix which makes it a "release candidate". Only when the release is being published, the version is changed to normal one (without the `-pre`) automatically.

With this scheme is quite easily to tell if we are running an official release, or not. Also - it seems to me that keeping both version and hash along-side at all times is useful anyway.